nml/ast/switch.py

```diff
@@ -62,7 +62,9 @@ def pre_process(self):
             seen_names.add(param.value)
 
         feature = next(iter(self.feature_set))
-        var_feature = action2var.get_feature(self)  # Feature of the accessed variables
+        var_scope = action2var.get_scope(feature, self.var_range)
+        if var_scope is None:
+            raise generic.ScriptError("Requested scope not available for this feature.", self.pos)
```

Comment:
separate issue to deal with, but most ScriptErrors do not have a trailing '.':
```
$ grep -R "raise generic.ScriptError" nml | grep '[.]"[^"]*$' | wc -l
131
$ grep -R "raise generic.ScriptError" nml | grep '[^.]"[^"]*$' | wc -l
284
```

Reply:
Most messages in this file have a trailing colon.